### PR TITLE
Refactor tests to JUnit 4

### LINKED ISSUE #174: Switch entirely to JUnit 4.x
Our test suites currently rely on a lot of JUnit 3 backwards compatibility leftovers in JUnit 4.x. It would be great to really truly move to 4.x.

### PR DESCRIPTION
**Pull request by acogoluegnes**
Fixes #174

### CODE REVIEW

File: src/test/java/com/rabbitmq/client/test/AbstractRMQTestSuite.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 import com.rabbitmq.tools.Host;
 
-public abstract class AbstractRMQTestSuite extends TestSuite {
+public abstract class AbstractRMQTestSuite { //extends TestSuite {
```
Comment (michaelklishin) [inline src/test/java/com/rabbitmq/client/test/AbstractRMQTestSuite.java]: We can remove this comment now, right?

Comment (acogoluegnes) [inline src/test/java/com/rabbitmq/client/test/AbstractRMQTestSuite.java]: Yes.

Comment (michaelklishin) [inline src/test/java/com/rabbitmq/client/test/AbstractRMQTestSuite.java]: I'll do it.

**Comment by michaelklishin** [comment]
@acogoluegnes please backport to `stable`. Thank you!

**Pull request merged**